backend/src/main/java/ca/gov/dtsstn/cdcp/api/service/UserService.java

```diff
@@ -134,4 +134,16 @@ private Predicate<ConfirmationCodeEntity> byCode(String code) {
 		return confirmationCode -> confirmationCode.getCode().equals(code);
 	}
 
+    public ConfirmationCodeStatus verifyConfirmationCode(ConfirmationCode confirmationCode, User user){
+        if (confirmationCode == null || confirmationCode.getCode().isBlank()) {
+            return ConfirmationCodeStatus.NO_CODE;
+        }
```

Comment:
We should throw early here (with an `Assert.hasText(..)`) rather than returning `NO_CODE` as we should always expect a confirmation code to be passed in order to verify it.

Reply:
I made a comment about using JSR validation for this, which is the standard approach when the incoming data is in the POST body. I will mark this comment as resolved and we can track the input validation via the other comment.